### PR TITLE
Add Table view in Read component

### PR DESCRIPTION
**Pull request by dtrucs**

### CODE REVIEW

File: src/modules/CRUD/components/Details/Read/Read.js [hunk 1]
```diff
@@ -13,6 +13,28 @@ const NO_FEATURE = 'CRUD.details.noFeature';
...
+const isTableObject = value => (
+  value.reduce((list, item) => {
+    if (!list.columns.length) {
+      return {
+        ...list,
+        columns: Object.keys(item),
+        rows: [Object.values(item)],
+      };
+    }
+    if (!Object.keys(item).sort().every((prop, i) => list.columns.sort()[i] === prop)) {
```
Comment (mabhub) [inline src/modules/CRUD/components/Details/Read/Read.js]: `if (whaaaaaats‽…) {`

Could you put this in a tiny pure function with a self explanatory name? _(which will probably take `item` & `list` as arguments)_


> « It's harder to read code than to write it »

File: src/modules/CRUD/components/Details/Read/Read.js [hunk 2]
```diff
@@ -13,6 +13,28 @@ const NO_FEATURE = 'CRUD.details.noFeature';
 
 const emptyStringNullOrUndef = value => ['', null, undefined].includes(value);
 
+const isTableObject = value => (
+  value.reduce((list, item) => {
+    if (!list.columns.length) {
```
Comment (mabhub) [inline src/modules/CRUD/components/Details/Read/Read.js]: I see three cases in this `reduce()` body. A simple line of comment above each case could easily explain them.

File: src/modules/CRUD/components/Details/Read/Read.js [hunk 3]
```diff
@@ -13,6 +13,37 @@ const NO_FEATURE = 'CRUD.details.noFeature';
...
+
+    // Fill the columns and rows arrays with the first iteration
+    if (!list.columns.length) {
+      return {
+        ...list,
+        columns,
+        rows: [rows],
+      };
+    }
+    // Define the object as no table
```
Comment (mabhub) [inline src/modules/CRUD/components/Details/Read/Read.js]:
```suggestion

    // If current iteration has different columns, it is not a table.
```

File: src/modules/CRUD/components/Details/Read/Read.js [hunk 4]
```diff
@@ -13,6 +13,37 @@ const NO_FEATURE = 'CRUD.details.noFeature';
...
 
+const haveStrictlySameValues = (arr1, arr2) =>
+  [...arr1].sort().join() === [...arr2].sort().join();
+
+const isTableObject = value => (
+  value.reduce((list, item) => {
+    const columns = Object.keys(item);
+    const rows = Object.values(item);
+
+    // Fill the columns and rows arrays with the first iteration
```
Comment (mabhub) [inline src/modules/CRUD/components/Details/Read/Read.js]:
```suggestion
    // If `list`does not have any columns (first iteration), fill array with current cols & rows
```

File: src/modules/CRUD/components/Details/Read/Read.js [hunk 5]
```diff
@@ -13,6 +13,37 @@ const NO_FEATURE = 'CRUD.details.noFeature';
...
+      };
+    }
+    // Define the object as no table
+    if (!haveStrictlySameValues(columns, list.columns)) {
+      return {
+        ...list,
+        isTable: false,
+      };
+    }
+    // Fill rows of each iteration
```
Comment (mabhub) [inline src/modules/CRUD/components/Details/Read/Read.js]:
```suggestion

    // Columns are the same: fill rows with current rows
```

**Comment by dtrucs** [comment]
Following our discussion, everything was refacto differently

**Pull request merged**